components/sup/tests/integration.rs

```diff
@@ -15,207 +20,630 @@ lazy_static! {
+                                              updated_files).await?;
+
+    Ok(())
+}
 
+// TODO: Improve this test case to check for restart cooldown once
+// we expose restart_attempts via the supervisor http gateway
+#[tokio::test]
+#[cfg_attr(feature = "ignore_integration_tests", ignore)]
+async fn restart_backoff_for_failed_init_hook() -> Result<()> {
+    let test_name = "restart_backoff_for_failed_init_hook";
+    let package_name = "config-and-hooks-no-reconfigure";
+    let applied_config = r#"
+    init_exit_code = 1
+    init_sleep = 5
+    "#;
     let origin_name = "sup-integration-test";
-    let package_name = "config-changes-hooks-do-not";
     let service_group = "default";
 
+    let service_min_backoff_period = Duration::from_secs(10);
+    let service_max_backoff_period = Duration::from_secs(20);
+    let service_restart_cooldown_period = Duration::from_secs(40);
+    let hab_root = utils::HabRoot::new(test_name);
+    let mut test_sup =
+        utils::TestSup::new_with_random_ports(&hab_root,
+                                              service_min_backoff_period,
+                                              service_max_backoff_period,
+                                              service_restart_cooldown_period).await?;
+
+    test_sup.start(Duration::from_secs(10)).await?;
+
     utils::setup_package_files(origin_name,
                                package_name,
                                service_group,
                                &FIXTURE_ROOT,
-                               &hab_root);
-
-    let mut test_sup =
-        utils::TestSup::new_with_random_ports(&hab_root, origin_name, package_name, service_group);
-
-    test_sup.start();
-    utils::sleep_seconds(3);
-
-    let pid_before_apply = hab_root.pid_of(package_name);
-    let hook_before_apply = hab_root.compiled_hook_contents(&package_name, "health-check");
-    let config_before_apply = hab_root.compiled_config_contents(&package_name, "config.toml");
-
-    test_sup.apply_config(
-                          r#"
-config_value = "applied"
-hook_value = "default"
-"#,
-    );
-    utils::sleep_seconds(2);
-
-    let pid_after_apply = hab_root.pid_of(package_name);
-    let hook_after_apply = hab_root.compiled_hook_contents(&package_name, "health-check");
-    let config_after_apply = hab_root.compiled_config_contents(&package_name, "config.toml");
-
-    assert_ne!(config_before_apply, config_after_apply);
-    assert_eq!(hook_before_apply, hook_after_apply);
-    assert_ne!(pid_before_apply, pid_after_apply);
+                               &hab_root).await?;
+
+    test_sup.ensure_service_started(package_name, service_group, Duration::from_secs(10))
+            .await?;
+
+    test_sup.service_stop(origin_name, package_name).await?;
+    test_sup.ensure_service_stopped(package_name, service_group, Duration::from_secs(10))
+            .await?;
+
+    test_sup.apply_config(package_name, service_group, applied_config)
```

Comment:
Yea, I had assumed that was the case. This test case no longer needs this though, because the init hook template change will cause a restart anyways now.